### PR TITLE
Test installation status when ideviceid is not installed

### PR DESCRIPTION
**Pull request by cbracken**
Ensure that flutter doctor returns a partial installation status when
ideviceid is not installed.

### CODE REVIEW

**Comment by cbracken** [comment]
@tvolkert @goderbauer @xster

**Comment by cbracken** [comment]
Spotted while fixing #10245

**Comment by tvolkert** [comment]
The newly added test fails on Travis and AppVeyor

**Comment by goderbauer** [comment]
Travis and Appveyor seem rather unhappy with this change :(

**Comment by xster** [comment]
Can you paste what the doctor outputs in that case? Just checking if we're explaining to the user what the various dependencies are for in case they fall into edge cases.

**Comment by cbracken** [comment]
@xster This doesn't fix #10245, this simply adds coverage where we had none before. I'll be fixing that in a separate patch.

**Comment by cbracken** [comment]
PTAL

**Comment by cbracken** [comment]
FYI this now *does* fix #10245. When `libimobiledevice` is at the incorrect version you now get this:
```
[-] iOS toolchain - develop for iOS devices (Xcode 8.3.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.2, Build version 8E2002
    ✗ libimobiledevice is incompatible with the installed Xcode version. To update, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
    ✗ ideviceinstaller not available; this is used to discover connected iOS devices.
      To install, run:
        brew update
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1
```

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 1]
```diff
@@ -33,6 +32,19 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
 
   bool get hasIDeviceId => exitsHappy(<String>['idevice_id', '-h']);
 
+  bool get hasWorkingLibimobiledevice {
+    // Verify that libimobiledevice tools are installed.
+    if (!hasIDeviceId)
+      return false;
+
+    // If a device is attached, verify that we can get its name.
+    final String result = runSync(<String>['idevice_id', '-l']);
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Can you use `run()` and make this return a `Future<bool>`?

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 2]
```diff
@@ -33,6 +32,19 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
...
+    if (result.isNotEmpty && !exitsHappy(<String>['idevicename']))
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: ... and `exitsHappyAsync()`

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: The entire library uses sync methods right now; will follow up with a patch that migrates everything over to the async equivalents.

**Pull request merged**